### PR TITLE
Tests: Bluetooth: Mesh: Add BLOB Server persistent storage BSim tests

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

**Comment by KKopyscinski** [comment]
A bit of RFC: in commit https://github.com/zephyrproject-rtos/zephyr/pull/56653/commits/b45405aedfe9e2e70e3dbc39bb298dff3d6ca6e0 I'm abandoning transfer that entered `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START` phase, but `XFER_START` message wasn't handled yet. `srv->state.xfer.size == 0` is used to verify if `XFER_START` was handled, as it should be set to some value if it did. Maybe we can just enter here `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START` and erase nothing, but I'm not sure. So alternative is:
```c
	if (srv->phase == BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START) {
		LOG_DBG("Transfer from 0x%04x (%llu) hasn't started yet", srv->state.cli,
			srv->state.xfer.id);
		return 0;
	} else {
		phase_set(srv, BT_MESH_BLOB_XFER_PHASE_SUSPENDED);
	}
```

**Comment by KKopyscinski** [comment]
These tests are passing locally but failing in CI, I'm converting to draft until I resolve this

**Comment by alxelax** [comment]
Hi @KKopyscinski, I really do not like the idea to add blob tests to the persistence test suite if they do not check anything related to the settings subsystem. 
Please create two test lists with and without persistence (how we've already done for provisioning tests. see sample).

**Comment by KKopyscinski** [comment]
> Hi @KKopyscinski, I really do not like the idea to add blob tests to the persistence test suite if they do not check anything related to the settings subsystem. Please create two test lists with and without persistence (how we've already done for provisioning tests. see sample).

When I resolve issue with passing this on CI (I think I got race somewhere) I'll split these from regular BLOB Cli tests

**Comment by KKopyscinski** [comment]
@PavelVPV @alxelax this is ready for review now. New tests are run with `pst` overlay and old ones run with default.

File: subsys/bluetooth/mesh/blob_srv.c
Comment (PavelVPV) [inline subsys/bluetooth/mesh/blob_srv.c:856]: We erase these variables also in `cancel`, `blob_srv_init` and `blob_srv_reset`. Could we join them into some common function?

Comment (PavelVPV) [inline subsys/bluetooth/mesh/blob_srv.c:856]: Not sure if it is correct to store something in settings subsystem while it loads the data. I'd suggest to either leave the stored value or handle this in `blob_srv_start`: https://github.com/zephyrproject-rtos/zephyr/blob/9ca9b91b1330d982ef24f610992682951ca85934/subsys/bluetooth/mesh/blob_srv.c#L885-L891

File: subsys/bluetooth/mesh/blob_srv.c
Comment (PavelVPV) [inline subsys/bluetooth/mesh/blob_srv.c:849]: Perhaps we should not store it here? https://github.com/zephyrproject-rtos/zephyr/blob/9ca9b91b1330d982ef24f610992682951ca85934/subsys/bluetooth/mesh/blob_srv.c#L914-L933

Or restore BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START phase?

Comment (KKopyscinski) [reply subsys/bluetooth/mesh/blob_srv.c:849]: I took this option, I restore `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START` phase in `blob_srv_settings_set`. This allows to continue as normal, no need to call `bt_mesh_blob_srv_recv` again, server is able to accept start normally.

File: tests/bsim/bluetooth/mesh/src/test_blob.c
Comment (Andrewpini) [inline tests/bsim/bluetooth/mesh/src/test_blob.c:1445]: Could this be handled by the default case? Is the default used for any expected cases? If not the BT_MESH_BLOB_XFER_PHASE_INACTIVE case could be removed both here and in the srv test case.

Comment (KKopyscinski) [reply tests/bsim/bluetooth/mesh/src/test_blob.c:1445]: yup, makes sens to me

File: subsys/bluetooth/mesh/blob_srv.c
Comment (alxelax) [inline subsys/bluetooth/mesh/blob_srv.c:852]: Should it be `LOG_WRN` since the process cannot be restored?

File: tests/bsim/bluetooth/mesh/src/test_blob.c
Comment (alxelax) [inline tests/bsim/bluetooth/mesh/src/test_blob.c:1568]: Seems handlers for `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_CHUNK` and `BT_MESH_BLOB_XFER_PHASE_COMPLETE` states are identical.

File: tests/bsim/bluetooth/mesh/src/test_blob.c
Comment (alxelax) [inline tests/bsim/bluetooth/mesh/src/test_blob.c:1601]: `break` is not necessary.

**Review by alxelax — CHANGES_REQUESTED** [review]

File: tests/bsim/bluetooth/mesh/src/test_blob.c
Comment (KKopyscinski) [reply tests/bsim/bluetooth/mesh/src/test_blob.c:1568]: It seems to be leftover from before I fixed client restoration - they were different at some point. I think I can make them fallthrough, I don't see reason to abstract this into funtion

File: subsys/bluetooth/mesh/blob_srv.c
Comment (KKopyscinski) [reply subsys/bluetooth/mesh/blob_srv.c:852]: In other words, state is restored, just transfer is not yet active. This is not an error, it's alternative path

File: tests/bsim/bluetooth/mesh/src/test_blob.c
Comment (KKopyscinski) [reply tests/bsim/bluetooth/mesh/src/test_blob.c:1573]: I abstracted this into functions, because they have small differences at the end. Same for both client and server parts

**Comment by KKopyscinski** [comment]
Seems like something broke in twister, but doesn't look like my code

**Comment by alxelax** [comment]
LGTM, when CI passes, I'll approve it.

**Comment by PavelVPV** [comment]
> Seems like something broke in twister, but doesn't look like my code

Will be fixed in https://github.com/zephyrproject-rtos/zephyr/pull/56813. Please rebase once it is merged.

**Comment by KKopyscinski** [comment]
> > Seems like something broke in twister, but doesn't look like my code
> 
> Will be fixed in #56813. Please rebase once it is merged.

Rebased, but seems like something more is still broken

**Comment by KKopyscinski** [comment]
@alxelax @PavelVPV Seems like CI is passing

**Review by alxelax — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**
